theano/tensor/opt.py

```diff
@@ -3502,6 +3507,18 @@ def local_adv_sub1_adv_inc_sub1(node):
             # investigate Alloc of 0s but with non constant shape.
             T.extract_constant(x, elemwise=False) != 0):
         return
+
+    if not inp.owner.op.set_instead_of_inc:
+        if config.warn.inc_subtensor1_opt:
+            warnings.warn(
+                'Your current code is fine, but Theano versions '
+                'between 0.7rc1 and 0.10 (or development versions '
+                'between Nov. 2014 and May 2017) '
+                'might have given incorrect results. '
+                'To disable this warning, set the Theano flag '
+                'warn.inc_subtensor1_opt to False.')
```

Comment:
Give more information. Otherwise people will fear too much problems. The problem only happen when there is duplicate indices. Add in the warning something like what is in the docstring.